nimble/host/src/ble_gap.c

```diff
@@ -1034,6 +1034,12 @@ ble_gap_master_failed(int status)
         ble_gap_master_connect_failure(status);
         break;
 
+    case BLE_GAP_OP_M_DISC:
+        STATS_INC(ble_gap_stats, initiate_fail);
+        ble_gap_disc_complete();
+        ble_gap_master_reset_state();
```

Comment:
could you double check if ` ble_gap_master_reset_state();` we need only for `BLE_GAP_OP_M_DISC:` and not for `BLE_GAP_OP_M_CONN' ?

Reply:
BTW - just noticed that above `ble_gap_master_failed(int status)` there is #if NIMBLE_BLE_CONNECT

I believe we need this #if inside that function. Also #if NIMBLE_BLE_SCAN should be there.

Reply:
> could you double check if ble_gap_master_reset_state(); we need only for BLE_GAP_OP_M_DISC: and not for `BLE_GAP_OP_M_CONN' ?

For `BLE_GAP_OP_M_CONN`, `ble_gap_master_connect_failure` --> `ble_gap_master_extract_state` --> calls `ble_gap_master_reset_state`. So we need it only for `BLE_GAP_OP_M_DISC`

Reply:
`#if NIMBLE_BLE_CONNECT` is applied for both functions: `ble_gap_master_failed` & `ble_gap_update_failed`, I will make changes accordinly. 

> I believe we need this #if inside that function. Also #if NIMBLE_BLE_SCAN should be there.

Yes, I will make the required changes.

Reply:
Hi @rymanluk I am getting CI failure

```
echo 'Building target=nordic_pca10056_bttester'

Building target=nordic_pca10056_bttester

+newt build -q -l info nordic_pca10056_bttester

2020/03/26 11:46:51.604 [INFO] Ignoring /home/travis/gopath/src/github.com/apache/mynewt-nimble/repos/apache-mynewt-core/hw/mcu/nordic/src/ext/nrfx/drivers/src/nrfx_power.c because package dictates it.

+rc=0

+[[ 0 -ne 0 ]]

+[[ 0 -eq 127 ]]

+for target in '${TARGETS}'

+echo 'Building target=nordic_pca10056_ext_advertiser'

Building target=nordic_pca10056_ext_advertiser

+newt build -q -l info nordic_pca10056_ext_advertiser

2020/03/26 11:47:43.511 [INFO] Ignoring /home/travis/gopath/src/github.com/apache/mynewt-nimble/repos/apache-mynewt-core/hw/mcu/nordic/src/ext/nrfx/drivers/src/nrfx_power.c because package dictates it.

Error: nimble/host/src/ble_gap.c:1028:1: error: 'ble_gap_master_failed' defined but not used [-Werror=unused-function]

 ble_gap_master_failed(int status)

 ^~~~~~~~~~~~~~~~~~~~~

cc1: all warnings being treated as errors

```

Reply:
I think it is better if we keep `#if NIMBLE_BLE_CONNECT` or `#if NIMBLE_BLE_SCAN` outside of `ble_gap_master_failed` function.